test/perf/PerfGemm.cpp

```diff
@@ -18,40 +22,68 @@
 
 #include <benchmark/benchmark.h>
 
+#include "include/OnnxMlirCompiler.h"
 #include "test/modellib/ModelLib.hpp"
+#include "test/perf/PerfHelper.hpp"
 
 using namespace std;
 
 const std::string modelName("./perfgemm");
 
-const CompilerOptionList opts{{onnx_mlir::OptionKind::CompilerOptLevel, "3"}};
-
 static void BM_MatmulSquare(benchmark::State &state) {
-  int IJK = state.range(0);
-  MatMul2DLibBuilder model(modelName, IJK, IJK, IJK);
-  assert(model.build() && model.compileAndLoad(opts) && model.prepareInputs() &&
+  int I = state.range(0);
+  int J = state.range(0);
+  int K = state.range(0);
+  MatMul2DLibBuilder model(modelName, I, J, K);
+  assert(model.build() && model.compileAndLoad() && model.prepareInputs() &&
          "failed matmul");
   for (auto _ : state)
     model.run();
-  state.SetComplexityN(IJK);
+  state.SetComplexityN(I);
+  PERF_RECORD_FLOPS(2.0 * I * J * K);
 }
 BENCHMARK(BM_MatmulSquare)
     ->RangeMultiplier(2)
-    ->Range(16, 1024)
+    ->Range(16, 2048)
+    ->Unit(benchmark::kMillisecond)
+    ->Complexity();
+
+static void BM_MatMulWithGemmSquare(benchmark::State &state) {
+  int I = state.range(0);
+  int J = state.range(0);
+  int K = state.range(0);
+  GemmLibBuilder model(modelName, I, J, K, false, false, 1, 1.0, 0.0);
+  assert(model.build() && model.compileAndLoad() && model.prepareInputs() &&
+         "failed gemm");
+  for (auto _ : state)
+    model.run();
+  state.SetComplexityN(I);
+  // Because alpha is 1, its not counted; beta is zero, sum of B is ignored.
+  PERF_RECORD_FLOPS(2.0 * I * J * K);
```

Comment:
Precisely, it would be `I * J * (K * 2 - 1)` where we have K multiplications and (K - 1) additions in the inner-most loop.

Reply:
good catch, will make the change.